content/anchor-frontend.md

```diff
@@ -0,0 +1,619 @@
+
+await sendTransaction(transaction, connection)
+```
+
+Similarly, you can use the same format to build an instruction using `.instruction` and then manually add the instructions to a new transaction. This builds a `TransactionInstruction` object using the instruction specified.
+
+```tsx
+// creates first instruction
+const instructionOne = await program.methods
+  .instructionOneName(instructionOneDataInputs)
+  .accounts({})
+  .instruction()
+
+// creates second instruction
+const instructionTwo = await program.methods
+  .instructionTwoName(instructionTwoDataInputs)
+  .accounts({})
+  .instruction()
+
+// add both instruction to one transaction
+const transaction = new Transaction().add(instructionOne, instructionTwo)
+
+// send transaction
+await sendTransaction(transaction, connection)
+```
+
+In summary, the Anchor `MethodsBuilder` provides a simplified and more flexible way to interact with on-chain programs. You can build an instruction, a transaction, or build and send a transaction using basically the same format without having to manually serialize or deserialize the accounts or instruction data.
+
+## Send Transactions
+
+As a refresher, we use the `sendTransaction` method from the `useWallet()` hook provided by `@solana/wallet-adapter-react` to send transactions with a wallet adapter.
+
+The `sendTransaction` method prompts a user to approve and sign a transaction before sending.
+
+```tsx
+import { useWallet } from "@solana/wallet-adapter-react"
+
+const { sendTransaction } = useWallet()
+```
+
+If the transaction was build using either `.instruction` or `.transaction` from the Anchor `MethodsBuilder`, you’ll need to send the transaction with `sendTransaction` to prompt a connected wallet for approval.
+
+```tsx
+sendTransaction(transaction, connection)
+```
+
+If the instruction requires additionally signatures, you need add the additionally signatures to `sendTransaction` using the following format:
+
+```tsx
+sendTransaction(transaction, connection, { signers: [] })
+```
+
+# Demo
+
+Let’s practice this together by building a frontend for the Counter program from last lesson. As a reminder, the Counter program has two instructions:
+
+- `initialize` - initializes a new `Counter` account and sets the `count` to `0`
+- `increment` - increments the `count` on an existing `Counter` account
+
+### 1. Download the starter code
+
+Download the starter code for this project [here](https://github.com/ZYJLiu/anchor-solana-ping-frontend/tree/starter). Once you have the starter code, take a look around. Install the dependencies with `npm install` and then run the app with `npm run dev`.
+
+This project is a simple Next.js application. It includes the `WalletContextProvider` we created in the Wallets lesson, the `idl.json` file for the Counter program, and the `Initialize` and `Increment` components we’ll be building throughout this demo. The `programId` of the program we’ll be invoking is also included in the starter code.
```

Comment:
Is this a reference to the Wallets lesson of the first course? Maybe add a link to it since these are two separate courses no guarantee that someone has gone through the other one